examples/digestauth.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * This example shows how to do Digest authentication.
+ * *
+ * @copyright Copyright (C) 2009-2015 fruux GmbH (https://fruux.com/).
+ * @author Markus Staab
+ * @license http://sabre.io/license/ Modified BSD License
+ */
+$userList = [
+    "user" => "password",
+    "user" => "password",
+];
+
+use Sabre\HTTP\Sapi;
+use Sabre\HTTP\Response;
+use Sabre\HTTP\Auth;
+
+// Find the autoloader
+$paths = [
+    __DIR__ . '/../vendor/autoload.php',
+    __DIR__ . '/../../../autoload.php',
+    __DIR__ . '/vendor/autoload.php',
+
+];
+
+foreach ($paths as $path) {
+    if (file_exists($path)) {
+        include $path;
+        break;
+    }
+}
+
+$request = Sapi::getRequest();
+$response = new Response();
+
+$digestAuth = new Auth\Digest("Locked down area", $request, $response);
+$digestAuth->init();
```

Comment:
It's basically there because I wrote it badly 7 years ago ;) 

https://github.com/fruux/sabre-dav/blob/0.6/lib/Sabre/HTTP/DigestAuth.php#L40

Not really happy with the auth classes at all anymore. If you look at how they are used within sabre/dav, you see they suck :P